app/models/topic_list.rb

```diff
@@ -138,7 +138,15 @@ def load_topics
       { category: :parent_category },
     ]
 
-    topic_preloader_associations.concat(DiscoursePluginRegistry.topic_preloader_associations.to_a)
+    DiscoursePluginRegistry.topic_preloader_associations.each do |a|
+      fields = a[:fields]
+      condition = a[:condition]
+      if condition.present?
+        topic_preloader_associations << fields if condition.present? && condition.call
```

Comment:
I don't think we need to check condition.present? a second time here, we are already in the branch checking for it